cmd/gardenlet/app/app.go

```diff
@@ -460,6 +467,36 @@ func (g *garden) cleanupOrphanedExtensionsServiceAccounts(ctx context.Context, g
 	return flow.Parallel(taskFns...)(ctx)
 }
 
+// TODO(Kostov6): Remove this code after v1.91 has been released.
+func (g *garden) cleanupGRMSecretFinalizers(ctx context.Context, seedClient client.Client, log logr.Logger) error {
+	var taskFns []flow.TaskFn
+
+	mrs := &resourcesv1alpha1.ManagedResourceList{}
+	if err := seedClient.List(ctx, mrs); err != nil {
+		return fmt.Errorf("failed to list ManagedResources while cleaning up GRM finalizers: %w", err)
+	}
+	for _, mr := range mrs.Items {
```

Comment:
You could try to reuse https://github.com/gardener/gardener/blob/0af0b2209fd9c12f7534aef97384e52c8d00c1a3/pkg/utils/kubernetes/client/client.go#L408